main.tf

```diff
@@ -29,6 +29,11 @@ locals {
   issue_labels_create    = var.issue_labels_create == null ? lookup(var.defaults, "issue_labels_create", local.issue_labels_create_computed) : var.issue_labels_create
   branch_protections_v3  = var.branch_protections_v3 == null ? var.branch_protections : var.branch_protections_v3
 
+  squash_merge_commit_title   = local.allow_squash_merge ? var.squash_merge_commit_title == null ? try(var.defaults.squash_merge_commit_title, null) : var.squash_merge_commit_title : null
+  squash_merge_commit_message = local.allow_squash_merge ? var.squash_merge_commit_title == null ? try(var.defaults.squash_merge_commit_title, null) : var.squash_merge_commit_title : null
+  merge_commit_title          = local.allow_merge_commit ? var.merge_commit_title == null ? try(var.defaults.merge_commit_title, null) : var.merge_commit_title : null
+  merge_commit_message        = local.allow_merge_commit ? var.merge_commit_message == null ? try(var.merge_commit_message, null) : var.merge_commit_message : null
+
```

Comment:
there is no condition when setting the fields.. we can always set them.

Reply:
nope that's not correct, as you can see in this test run, `allow_squash_merge` xor `allow_merge_commit` need to be set to `true` for setting those values otherwise the API returns an error https://github.com/mineiros-io/terraform-github-repository/actions/runs/3394517816/jobs/5643205775#step:4:1092

Reply:
the error is based on typos in the variable names.

Reply:
Not sure if that's correct, did you see the error

>  Error: POST https://api.github.com/orgs/***/repos: 422 Repository creation failed. [{Resource:Repository Field: Code:custom Message:Sorry, you need to allow the squash merge strategy in order to set the default squash commit message title or message. (no_squash_merge_strategy)}]

Reply:
comitted suggested changes to test... works ;) .... conflicts would need to be resolves and default set to `null` again... or remove the complex default handling as it's deprectaed now anyway.. ;)